src/rcore.c

```diff
@@ -3163,28 +3165,37 @@ bool IsKeyUp(int key)
     return up;
 }
 
-// Get the last key pressed
-int GetKeyPressed(void)
+// Get the last key and scancode pressed
+int GetKeyPressedPro(int* scanCode)
 {
     int value = 0;
 
     if (CORE.Input.Keyboard.keyPressedQueueCount > 0)
     {
         // Get character from the queue head
-        value = CORE.Input.Keyboard.keyPressedQueue[0];
+        value = CORE.Input.Keyboard.keyPressedQueue[0].keycode;
+        if (scanCode != NULL)
+            *scanCode = CORE.Input.Keyboard.keyPressedQueue[0].scancode;
 
         // Shift elements 1 step toward the head
         for (int i = 0; i < (CORE.Input.Keyboard.keyPressedQueueCount - 1); i++)
             CORE.Input.Keyboard.keyPressedQueue[i] = CORE.Input.Keyboard.keyPressedQueue[i + 1];
 
         // Reset last character in the queue
-        CORE.Input.Keyboard.keyPressedQueue[CORE.Input.Keyboard.keyPressedQueueCount - 1] = 0;
+        CORE.Input.Keyboard.keyPressedQueue[CORE.Input.Keyboard.keyPressedQueueCount - 1].keycode = 0;
+        CORE.Input.Keyboard.keyPressedQueue[CORE.Input.Keyboard.keyPressedQueueCount - 1].scancode = -1;
         CORE.Input.Keyboard.keyPressedQueueCount--;
     }
 
     return value;
 }
 
+// Get the last key pressed
+int GetKeyPressed(void)
+{
+    return GetKeyPressedPro(NULL);
```

Comment:
Considering `GetKeyPressed()` is very short, I think code lines can be "duplicated" for `GetKeyPressedRaw()`.